subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/provider/ClassPathModeExceptionCollector.kt

```diff
@@ -16,15 +16,16 @@
 package org.gradle.kotlin.dsl.provider
 
 import org.gradle.internal.concurrent.Stoppable
+import org.gradle.internal.service.scopes.ExceptionCollector
 
 
-open class ClassPathModeExceptionCollector : Stoppable {
+open class ClassPathModeExceptionCollector(val suppressor: ExceptionCollector) : Stoppable {
 
     private
     val collection = mutableListOf<Exception>()
 
     val exceptions: List<Exception>
```

Comment:
Since this is a public API I could not move it to a single service.

Reply:
It is not public API, but it is effectively now used by the Kotlin Gradle Plugin so we need to keep it around a little while.

Another approach would have been to keep this service with it's current API but change it so that it delegates to a new actual implementation.

Reply:
I went with the delegation approach.